e2e/utils/openApp.ts

```diff
@@ -0,0 +1,53 @@
+const appConfig = require('../../app.config.js');
+const { resolveConfig } = require('detox/internals');
+
+const platform = device.getPlatform();
+
+module.exports.openApp = async function openApp() {
+  const { configurationName } = await resolveConfig();
+  if (configurationName.split('.')[1] === 'debug') {
+    return await openAppForDebugBuild(platform);
+  } else {
+    return await device.launchApp({
+      newInstance: true,
+    });
+  }
+};
+
+async function openAppForDebugBuild(platform: string) {
+  const deepLinkUrl = process.env.EXPO_USE_UPDATES
+    ? // Testing latest published EAS update for the test_debug channel
+      getDeepLinkUrl(getLatestUpdateUrl())
+    : // Local testing with packager
+      getDeepLinkUrl(getDevLauncherPackagerUrl(platform));
+
+  if (platform === 'ios') {
+    await device.launchApp({
+      newInstance: true,
+    });
+    sleep(3000);
+    await device.openURL({
+      url: deepLinkUrl,
+    });
+  } else {
+    await device.launchApp({
+      newInstance: true,
+      url: deepLinkUrl,
+    });
+  }
+
+  await sleep(3000);
+}
+
+const getDeepLinkUrl = (url: string) =>
+  `exp+graasp-mobile-builder://expo-development-client/?url=${encodeURIComponent(url)}`;
+
+const getDevLauncherPackagerUrl = (platform: string) =>
+  `http://localhost:8081?platform=${platform}&dev=true&minify=false&disableOnboarding=1`;
```

Comment:
Why `8081`? Can this be an env variable? Or a constant for test?
In fact I don't know what that link is used for.

Reply:
It's the Expo CLI Metro Server default port. I believe the only way of changing it is by specifying it in the `yarn start -dev-client` command
https://docs.expo.dev/more/expo-cli/